src/hotspot/share/opto/loopnode.cpp

```diff
@@ -437,77 +439,71 @@ Node* PhaseIdealLoop::loop_exit_control(Node* x, IdealLoopTree* loop) {
   return iftrue;
 }
 
-Node* PhaseIdealLoop::loop_exit_test(Node* back_control, IdealLoopTree* loop, Node*& incr, Node*& limit, BoolTest::mask& bt, float& cl_prob) {
-  Node* iftrue = back_control;
+PhaseIdealLoop::LoopExitTest PhaseIdealLoop::loop_exit_test(const Node* back_control, const IdealLoopTree* loop) {
```

Comment:
Just an idea here: Could this also be part of `LoopExitTest` instead? Then a user could do something like:
```
LoopExitTest loop_exit_test(...);
 // i.e. = PhaseIdealLoop::loop_exit_test() but with a `_is_valid` flag. Then at the end you 
 // can also check for the right Cmp opcode and that it's not null which the current caller of 
 // loop_exit_test() are all doing. If that's off, you set `_is_valid` to false accordingly.
loop_exit_test.build();
if (loop_exit_test.is_not_valid()) {
...
}

```